File: DESKTOP/VP-Albi_Zrt_DESKTOP/VP-Albi_Zrt_DESKTOP/Model/Task.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VP_Albi_Zrt_DESKTOP.Model
{
    public class Task
    {
        public enum eStatus { Open, Acceptence, In_work, Done };
        public enum eAcceptanceProperty { Accepted, Accepted_with_conditions, Denied, Waiting_for_reply };

        public int ID { get; set; }
        public DateTime CreationDate { get; set; }
        public User From { get; set; }
        public User To { get; set; }
        public string Description { get; set; }
        public eStatus TaskStatus { get; set; }
        public DateTime RequestedCompletionDate { get; set; }
        public DateTime PlannedCompletionDate { get; set; }
        public eAcceptanceProperty AcceptanceProperty { get; set; }
        public string AcceptanceMessage { get; set; }

        
        public Task() { }
    }
}

File: WEB/albiapp/app/backend/models/Task.java
package backend.models;


import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Entity
public class Task {
    public enum TaskStatus implements NamedEnum {
        OPEN("Open"),
        ACCEPTANCE("Acceptance"),
        IN_WORK("In work"),
        DONE("Done");

        private final String name;

        TaskStatus(String name) {
            this.name = name;
        }

        @Override
        public String getName() {
            return name;
        }
    }

    public enum AcceptanceStatus implements NamedEnum {
        ACCEPTED("Accepted"),
        ACCEPTED_WITH_CONDITIONS("Accepted with conditions"),
        DENIED("Denied"),
        WAITING_FOR_REPLY("Waiting for reply");

        private final String name;

        AcceptanceStatus(String name) {
            this.name = name;
        }

        @Override
        public String getName() {
            return name;
        }
    }

    @Id
    @GeneratedValue
    public Long id;

    public Date creationDate = new Date();

    @ManyToOne
    @NotNull
    public FlatUser from;

    @ManyToOne
    @NotNull
    public FlatUser to;

    @NotNull
    public String name;

    public String description;

    @NotNull
    public TaskStatus taskStatus;

    public Date requestedCompletionDate;

    public Date plannedCompletionDate;

    @NotNull
    public AcceptanceStatus acceptanceStatus;

    public String acceptanceMessage;

    public Task(FlatUser from, FlatUser to, String description, TaskStatus taskStatus,
                Date plannedCompletionDate, String acceptanceMessage) {
        this.from = from;
        this.to = to;
        this.description = description;
        this.taskStatus = taskStatus;
        this.plannedCompletionDate = plannedCompletionDate;
        this.acceptanceMessage = acceptanceMessage;
    }

    public Task() {
    }
}